student-info-system/student/views.py

```diff
@@ -139,33 +139,44 @@ def history(request):
     data = {
         'auser': the_user,
     }
-    data.update(
-        filtered_table2(
-            name='history',
-            qs=the_user.profile.student.course_history(),
-            filter=StudentHistoryFilter,
-            table=StudentHistoryTable,
-            request=request,
-            wrap_list=False,
-            self_url=reverse('student:history'),
-            click_url=reverse('schooladmin:sectionstudent', args=[DUMMY_ID]),
-        ))
-    data.update(
-        filtered_table2(
-            name='semesters',
-            qs=the_user.profile.student.semesters.all(),
-            filter=SemesterFilter,
-            table=SemestersSummaryTable,
-            request=request,
-            self_url=reverse('student:history'),
-            click_url=reverse('schooladmin:semester', args=[DUMMY_ID]),
-        ))
+    ssects = ssects_by_sem(the_user)
+    sem_gpas = []
+    # lets get the gpa for each semester
+    for sem in ssects:
+        credits_attempted = grade_pnts = 0
+        for ssec in sem:
+            credits_attempted = credits_attempted + ssec.section.course.credits_earned
+            grade_pnts = grade_pnts + ssec.grade_points
+        gpa = grade_pnts / float(credits_attempted)
+        sem_gpas.append(gpa)
+
+    ssects_gpas = zip(sem_gpas, ssects)
+    data['ssects_gpas'] = ssects_gpas
+    #    data.update(
+    #        filtered_table2(
+    #            name='history',
+    #            qs=the_user.profile.student.course_history(),
+    #            filter=StudentHistoryFilter,
+    #            table=StudentHistoryTable,
+    #            request=request,
+    #            wrap_list=False,
+    #            self_url=reverse('student:history'),
+    #            click_url=reverse('schooladmin:sectionstudent', args=[DUMMY_ID]),
+    #        ))
+    #    data.update(
+    #        filtered_table2(
+    #            name='semesters',
+    #            qs=the_user.profile.student.semesters.all(),
+    #            filter=SemesterFilter,
+    #            table=SemestersSummaryTable,
+    #            request=request,
+    #            self_url=reverse('student:history'),
+    #            click_url=reverse('schooladmin:semester', args=[DUMMY_ID]),
+    #        ))
 
     remaining = the_user.profile.student.requirements_met_list()
     stats = remaining.filter(met=False).aggregate(remaining_course_count=Count('id'),
                                                   remaining_credit_count=Sum('credits_earned'))
-    if stats['remaining_credit_count'] is None:
```

Comment:
er, please merge this in. It keeps the remaing credit count from being "None"

Reply:
done